processing/results_analysis/gpkg_conversion.py

```diff
@@ -0,0 +1,158 @@
+        if gpkg_path.endswith(".file"):
+            gpkg_path = gpkg_path.rsplit(".", 1)[0] + ".gpkg"
+
+        s = QgsSettings()
+        s.setValue("threedi-results-analysis/gridadmin_to_gpkg/last_input", gridadmin_folder)
+        s.setValue("threedi-results-analysis/gridadmin_to_gpkg/last_output_gpkg", gpkg_path)
+        plugin_dir = os.path.dirname(os.path.realpath(__file__))
+        styles_dir = os.path.join(plugin_dir, "styles")
+
+        progress = Progress(feedback)
+        exporter = GeopackageExporter(input_gridadmin, gpkg_path)
+        exporter.export(progress.update)
+        feedback.pushInfo("Export done!")
+
+        # Unfortunately, temporaryLayerStore keeps layers to be added as a dictionary, so the order is lost
+        data_srcs = OrderedDict(
+            [
+                ("Obstacle", "obstacle"),
+                ("Cell", "cell"),
+                ("Pump (point)", "pump"),
+                ("Pump (line)", "pump_linestring"),
+                ("Node", "node"),
+                ("Flowline", "flowline"),
+            ]
+        )
+
+        layers = dict()
+        empty_layers = []
+        epsg_codes = set()
+        for layer_name, table_name in data_srcs.items():
+            uri = gpkg_path + f"|layername={table_name}"
+            layer = QgsVectorLayer(uri, layer_name, "ogr")
+            layers[layer_name] = layer
+
+            # only load layers that contain some features
+            if not layers[layer_name].featureCount():
+                empty_layers.append(layer_name)
+                continue
+
+            # apply the style and add for loading when alg is completed
+            qml_path = safe_join(styles_dir, f"{table_name}.qml")
+            if os.path.exists(qml_path):
+                layer.loadNamedStyle(qml_path)
+            context.temporaryLayerStore().addMapLayer(layer)
+            layer_details = QgsProcessingContext.LayerDetails(layer_name, context.project())
+            layer_details.forceName = True
+            context.addLayerToLoadOnCompletion(layer.id(), layer_details,)
+
+        # Empty layers info
+        if empty_layers:
+            empty_info = "\n\nThe following layers contained no feature:\n * " + "\n * ".join(empty_layers) + "\n\n"
+            feedback.pushInfo(empty_info)
+
+        # Set project CRS only if all source layers have the same CRS
+        if len(epsg_codes) == 1:
+            code_int = int(list(epsg_codes)[0])
+            crs = QgsCoordinateReferenceSystem.fromEpsgId(code_int)
+            if crs.isValid():
+                context.project().setCrs(crs)
+                crs_info = "Setting project CRS according to the source gridadmin file."
+            else:
+                crs_info = "Skipping setting project CRS - does gridadmin file contains a valid EPSG code?"
+        else:
+            crs_info = (
+                f"Skipping setting project CRS - the source file {input_gridadmin}" " contained data with various CRS."
```

Comment:
This is one of those long lines, even with an implicitly continued string, ready for shortening. => `make beautiful`

Reply:
That kind of a bit weird formatting was done by black. I rephrased that message anyway. :)